### PR TITLE
Blog app - controllers specs

### PR DESCRIPTION
**Pull request by AlselwiAisha**
### Project requirements

- Add RSpec as a gem to your project.
- Create a Request spec file for all your controllers.
- Make sure that for each action you check:
      - If response status was correct.
      - If a correct template was rendered.
      - If the response body includes correct placeholder text.

### CODE REVIEW

File: spec/request/users_spec.rb [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+  describe 'GET #index' do
+    it 'Response status is correct' do
+      get '/users'
+      expect(response).to have_http_status(:ok)
+    end
+
+    it 'Returns a successful response for index' do
+      get '/users'
+      expect(response).to be_successful
+    end
```
Comment (omar-muhamad) [inline spec/request/users_spec.rb]: - [ ] As per this [requirement](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/controller_specs_project.md#:~:text=If%20a%20correct%20template%20was%20rendered.), This test should check If a correct template was rendered, not the condition of the response as you did. Please fix that.
```suggestion
    it 'Returns a successful response for index' do
      get '/users'
      expect(response).to render_template(:index)
    end
```

File: spec/request/users_spec.rb [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+  describe 'GET #show' do
+    it 'Response status is correct' do
+      get '/users/:id'
+      expect(response).to have_http_status(:ok)
+    end
+
+    it 'Returns a successful response for show' do
+      get '/users/:id'
+      expect(response).to be_successful
+    end
```
Comment (omar-muhamad) [inline spec/request/users_spec.rb]: - [ ] Same here, As per this [requirement](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/controller_specs_project.md#:~:text=If%20a%20correct%20template%20was%20rendered.), This test should check If a correct template was rendered, not the condition of the response as you did. Please fix that.

File: spec/request/posts_spec.rb [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+  describe 'GET #index' do
+    it 'Response status is correct' do
+      get '/users/:id/posts'
+      expect(response).to have_http_status(:ok)
+    end
+
+    it 'Returns a successful response for index' do
+      get '/users/:id/posts'
+      expect(response).to be_successful
+    end
```
Comment (omar-muhamad) [inline spec/request/posts_spec.rb]: - [ ] Same here, As per this [requirement](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/controller_specs_project.md#:~:text=If%20a%20correct%20template%20was%20rendered.), This test should check If a correct template was rendered, not the condition of the response as you did. Please fix that.

File: spec/request/posts_spec.rb [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+  describe 'GET #show' do
+    it 'Response status is correct' do
+      get '/users/:id/posts/:id'
+      expect(response).to have_http_status(:ok)
+    end
+
+    it 'Returns a successful response for show' do
+      get '/users/:id/posts/:id'
+      expect(response).to be_successful
+    end
```
Comment (omar-muhamad) [inline spec/request/posts_spec.rb]: - [ ] Same here, As per this [requirement](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/controller_specs_project.md#:~:text=If%20a%20correct%20template%20was%20rendered.), This test should check If a correct template was rendered, not the condition of the response as you did. Please fix that.

**Review by omar-muhamad — CHANGES_REQUESTED** [review]
# Changes Required ♻️
### Hi @AlselwiAisha,

### Nice job, you are almost there.

cat-driving-serious

## (Highlights) Good Points: :+1:
- Your work is well documented. ✔
- The correct workflow is used.  ✔
- All linters checks are ok. ✔

## (Changes Required) Aspects to improve: ♻️

- [ ] Please check the inline comments below ⬇️.

## [Optional] suggestions:

- Nothing to mention.

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

**Cheers and Happy coding!👏👏👏**

Feel free to leave any questions or comments in the PR thread if something is not 100% clear, and please remember to tag me in your question so I can receive the notification**

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Review by Gambit142 — APPROVED** [review]
Hi @AlselwiAisha 👋,

## Status: Approved 🟢   

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉

### Highlights
- Setup linters correctly ✔️ 
- Use of `Rspec` testing Library ✔️ 
- Wrote request specs for Users and Posts controllers ✔️ 
- All tests pass ✔️ 

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

#### Please tag me in your question so I can receive the notification. You can tag me by typing @Gambit142

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**